include/rcutils/process.h

```diff
@@ -54,6 +70,65 @@ RCUTILS_PUBLIC
 RCUTILS_WARN_UNUSED
 char * rcutils_get_executable_name(rcutils_allocator_t allocator);
 
+/// Information about a subprocess created by this process
+typedef struct rcutils_process_s
+{
+#if defined _WIN32 || defined __CYGWIN__
+  /// The open handle to the process.
+  HANDLE handle;
+#endif
+
+  /// The process ID of the process.
+  int pid;
```

Comment:
i think we could use `pid_t` here for non-windows system, and technically that is the correct one. (i understand that is typically signed int.)

```suggestion
  #include <sys/types.h>
  /// The process ID of the process.
  pid_t pid;
```

Reply:
I actually bounced back and forth on this one. We already have a `rcutils_get_pid` function in this file and it returns an `int`. If I have to choose, I'd rather be consistent, but I'm open to differing opinions.

I'll also note that `pid_t` isn't defined on Win32, so we'd probably have to make this part of the `#ifdef`.

Reply:
i am good to go with yours. thanks!